opm/input/eclipse/Schedule/MSW/WelSegsSet.cpp

```diff
@@ -58,6 +58,24 @@ WelSegsSet::difference(const std::set<std::string>& compsegs,
     return difference;
 }
 
+std::vector<WelSegsSet::Entry> 
+WelSegsSet::intersection(const std::set<std::string>& wells1,
+                         const std::set<std::string>& wells2) const
+{
+    std::vector<Entry> intersection;
+    intersection.reserve(entries_.size());
+
+    for (const auto& entry : entries_) {
+        const auto& name = std::get<0>(entry);
+        if (wells1.find(name) != wells1.end() && wells2.find(name) != wells2.end()) {
```

Comment:
You can use `std::set_intersection` to get the sorted common between `wells1` and `wells2`.  If there is no common, I think you do not need to check `entries_`. 

is it possible to have wells inside `wells1` and `wells2`, while it is not in the `entries_`? if not, we even do not need to check `entries`?